tensorflow/core/platform/env.cc

```diff
@@ -361,6 +374,32 @@ Status WriteStringToFile(Env* env, const string& fname,
   return s;
 }
 
+Status CopyFile(FileSystem* src_fs, const string& src, FileSystem* target_fs,
+                const string& target) {
+  uint64 size;
+  TF_RETURN_IF_ERROR(src_fs->GetFileSize(src, &size));
+
+  std::unique_ptr<RandomAccessFile> src_file;
+  TF_RETURN_IF_ERROR(src_fs->NewRandomAccessFile(src, &src_file));
```

Comment:
Do we want copies to be atomic? I just noticed our GCS RandomAccessFile impl doesn't lock-in the [generation](https://cloud.google.com/storage/docs/generations-preconditions). Can we do that @dbcode? If that can happen soon, I don't see any reason to block this PR.